WordPress/Classes/ViewRelated/Me/All Domains/Coordinators/AllDomainsAddDomainCoordinator.swift

```diff
@@ -0,0 +1,31 @@
+import Foundation
+
+@objc final class AllDomainsAddDomainCoordinator: NSObject {
+    static func presentAddDomainFlow(in allDomainsViewController: AllDomainsListViewController) {
+        let coordinator = RegisterDomainCoordinator(site: nil)
+        let domainSuggestionsViewController = RegisterDomainSuggestionsViewController.instance(
+            coordinator: coordinator,
+            domainSelectionType: .purchaseFromDomainManagement,
+            includeSupportButton: false,
+            title: Strings.searchTitle)
+
+
+        let domainPurchasedCallback = { (domainViewController: UIViewController, domainName: String) in
+            allDomainsViewController.reloadDomains()
+        }
```

Comment:
[Suggestion] @hassaanelgarem This closure is capturing `allDomainsViewController`. I'm not sure if that's going to cause a memory leak or not, what do you think of adding [weak allDomainsViewController] to be safe?

Feel free to ignore this suggestion if you think this is not going to a cause a memory leak.

Reply:
Thanks for raising this! I don't think it's needed because `allDomainsViewController` doesn't have a strong reference to this closure. I've confirmed this using the Memory graph 👍